code/parachain/runtime/picasso/src/lib.rs

```diff
@@ -178,6 +196,78 @@ parameter_types! {
 	pub const SS58Prefix: u8 = 49;
 }
 
+#[derive(Clone, Debug, Eq, PartialEq, Default)]
+pub struct Router {}
+
+impl ModuleRouter for Router {
+	fn get_route_mut(&mut self, _module_id: &ModuleId) -> Option<&mut dyn Module> {
+		None
+	}
+
+	fn has_route(_module_id: &ModuleId) -> bool {
+		false
+	}
+
+	fn lookup_module_by_port(_port_id: &PortId) -> Option<ModuleId> {
+		None
+	}
+}
+
+pub struct IbcDenomToAssetIdConversion;
+
+impl DenomToAssetId<Runtime> for IbcDenomToAssetIdConversion {
+	type Error = DispatchError;
+
+	fn from_denom_to_asset_id(denom: &String) -> Result<CurrencyId, Self::Error> {
+		let denom_bytes = denom.as_bytes().to_vec();
+		if let Some(id) = IbcDenoms::<Runtime>::get(&denom_bytes) {
+			return Ok(id)
+		}
+
+		let asset_id =
+			<currency_factory::Pallet<Runtime> as CurrencyFactoryT>::create(RangeId::IBC_ASSETS)?;
+
+		IbcDenoms::<Runtime>::insert(denom_bytes.clone(), asset_id);
+		IbcAssetIds::<Runtime>::insert(asset_id, denom_bytes);
+
+		let location = XcmAssetLocation::new(MultiLocation::new(
```

Comment:
Isn't this just `HERE` @dzmitry-lahoda

Reply:
i would perfer we do not register anything in assets registry until rewrite. imho. we could do it in separate pr also. 

there could be different encoding depending on if we use or not xcm v3 i  think.

on current xcm v2 code base, it would be (parent = 0, general index = asset id).  or better (parent = 0, pallet id = 190, general index).

Reply:
pallet id convention helps to identify protocol which manages tokens on this chain.

Reply:
yeah

Better hardcode information and do not have any registration. as it will register new token each transfer? 

we will handle it later.

Reply:
I'd rather us treat the IBC Denom and the XCM location as different types of foreign asset ID.